dataflow/src/main/java/org/checkerframework/dataflow/expression/JavaExpression.java

```diff
@@ -911,4 +965,18 @@ private static boolean isVarArgsInvocation(
     return TypesUtils.getArrayDepth(ElementUtils.getType(lastParamElt))
         != TypesUtils.getArrayDepth(lastArgType);
   }
+
+  /**
+   * Given a method call sequence (e.g., m.foo(p1).bar().baz(p2)), determine whether all the
+   * arguments are unassignable.
+   *
+   * @param methodCallSequence the method call sequence to extract arguments from
+   * @return true if all the arguments in the method call sequence are unassignable
+   */
+  public static boolean areAllArgumentsUnassignable(JavaExpression methodCallSequence) {
```

Comment:
In general, negations reduce readability.  So I suggest changing this to `isAnyArgumentAssignable`, and adjusting the implementation to match.